location_extractor/extractor.py

```diff
@@ -0,0 +1,188 @@
+
+            regions_in_country = [region for region in potential_regions if region.country in countries]
+            regions_on_continents = [region for region in potential_regions if region.country.continent in continents]
+
+            if regions_in_country:
+                regions = regions.union(regions_in_country)
+            elif regions_on_continents:
+                regions = regions.union(regions_on_continents)
+            elif potential_regions:
+                regions = regions.union(potential_regions)
+            else:
+                remaining_places.add(place)
+
+        return list(regions), remaining_places
+
+    def get_cities(self, places: Set[str], continents: List[Continent], countries: List[Country],
+                   regions: List[Region]) -> Tuple[List[City], Set[str]]:
+        remaining_places = set()
+        cities = set()
+        for place in places:
+            query = f'city_name == "{capitalize_name(remove_accents(place))}"'
+            cities_df = self.locations_data.query(query)
+
+            potential_cities = City.from_dicts(cities_df.to_dict("records"))
+
+            cities_in_regions = [city for city in potential_cities if city.region in regions]
+            cities_in_country = [city for city in potential_cities if city.country in countries]
+            cities_on_continents = [city for city in potential_cities if city.country.continent in continents]
+
+            if cities_in_regions:
+                cities = cities.union(cities_in_regions)
+            elif cities_in_country:
+                cities = cities.union(cities_in_country)
+            elif cities_on_continents:
+                cities = cities.union(cities_on_continents)
+            elif potential_cities:
+                cities = cities.union(potential_cities)
+            else:
+                remaining_places.add(place)
+
+        return list(cities), remaining_places
+
+    def find_acronym(self, name):
+        name_clean = remove_accents(name).upper()
+        name_clean = re.sub(r"(^|\s)(the)(\s|$)", "", name_clean, flags=re.IGNORECASE)
+        if name_clean == "UK":
+            name_clean = "GB"
+        elif name_clean == "USA":
+            name_clean = "US"
+        query = f"country_iso_code == '{name_clean}'"
+        countries_df = self.locations_data.query(query)
+        if not countries_df.empty:
+            return countries_df['country_name'].drop_duplicates().iloc[0]
+
+    def is_a_country(self, name: str) -> bool:
+        countries, remaining_places = self.get_countries([name], [])
+        return name not in remaining_places
+
+    def is_location(self, place: str) -> bool:
+        continents, countries, regions, cities = self.find_locations([place])
+
+        if cities or regions or countries:
+            return True
+        else:
+            return False
+
+    def clean_sublocations(self, places):
+        pattern = re.compile(r"(west|south|north|east)(ern)?", re.IGNORECASE)
```

Comment:
let's move this pattern to class attribute, so it will be compiled only once

Also let's compile all regexp touched in this PR and make them class attributes or some constants